Add option to customize template delimiters

kuguma opened the pull request:
## Motivation

In the MQTT integration I had to author many `value_template`s. Heavy use of escape sequences made templates hard to read, so I added a way to change the Jinja delimiters.

https://www.home-assistant.io/integrations/template/

## What’s changed

* Added an option to change template delimiters. It is configurable from the Home Assistant UI. Default behavior is unchanged.
* Replaced `j2cli` with `jinjanator` as the rendering backend because `j2cli` does not support custom delimiters.

## Backward compatibility

* No breaking changes. Existing templates using the default delimiters continue to work.
* The new option is opt-in.

kuguma committed:
Update repository details for Jinja2Config fork

claude committed:
Add custom Jinja2 delimiter options and migrate to j2cli

- Add configuration options for all Jinja2 delimiters:
  - variable_start_string/variable_end_string
  - block_start_string/block_end_string
  - comment_start_string/comment_end_string
- Migrate from jinja-cli to j2cli for better customization support
- Generate dynamic customization file based on addon config
- Update README with configuration documentation and examples
- Add test files to verify delimiter customization works
- Bump version to 2.0.0 (breaking change)

This allows users to customize Jinja2 delimiters to avoid conflicts
with other templating systems or syntax requirements.

kuguma committed:
Merge pull request #3 from kuguma/claude/add-jinja-delimiter-options-011CUu8GxCEzpnzruvLHrdPh

Add custom Jinja2 delimiter options and migrate to j2cli

claude committed:
Migrate from j2cli to jinjanator for Python 3.12+ support

- Replace j2cli with jinjanator (actively maintained successor)
- Jinjanator supports Python 3.12+ (j2cli was archived in June 2024)
- Update error messages and documentation to reflect the change
- All tests pass with jinjanator 25.3.0
- No breaking changes to functionality or configuration

The jinjanator package provides the same 'j2' CLI entrypoint as j2cli
and supports the same --customize option, making the migration seamless.

claude committed:
Bump version to 2.0.1 for jinjanator migration

kuguma committed:
Merge pull request #4 from kuguma/claude/add-jinja-delimiter-options-011CUu8GxCEzpnzruvLHrdPh

Migrate from j2cli to jinjanator for Python 3.12+ support

kuguma committed:
delete claude's test files

kuguma committed:
delete unnecessary changes

kuguma committed:
rollback repository.json

kuguma commented:
I’m using this in my environment, and it’s been working correctly so far.

Please decide the version number as appropriate. @tonyroberts

tonyroberts commented:
Thanks @kuguma, I will take a look when I get some time

tonyroberts commented on config.yaml:
```diff
@@ -14,6 +14,18 @@ arch:
 options:
   log_level: debug
   config_dir: "/config"
+  variable_start_string: "{{"
```
It would be better if these didn't have defaults here, then they wouldn't always be visible in the config (the user would see them if 'Show unused optional configuration options' was selected, but otherwise only the basic options would be shown).

tonyroberts commented on rootfs/usr/bin/jinja2config.sh:
```diff
@@ -12,6 +12,46 @@ if ! type prettier > /dev/null 2>&1; then
+    params = {}
+
+    # Read delimiter settings from environment variables
+    var_start = os.environ.get('J2_VAR_START', '{{')
```
No need for defaults here since you're testing 'if value' below, so either the default or the test is redundant

tonyroberts reviewed (dismissed):
Couple of minor changes requested. Would also benefit from a translation file for these new options.

tonyroberts committed:
Remove defaults for j2 options and add translations

tonyroberts committed:
Move j2 customizations to a template file

It's clearer than writting it as a temp file from the script.

Also if options are set, let j2 use its defaults instead of hardcoding
them in the script.

Log the customizations if trace logging is enabled for easier trouble
shooting.

tonyroberts committed:
Revert change to version number

That will be done on the main branch prior to release.

tonyroberts commented:
Thanks for the PR @kuguma
